src/vs/base/parts/quickinput/browser/quickInput.ts

```diff
@@ -516,9 +516,30 @@ class QuickPick<T extends IQuickPickItem> extends QuickInput implements IQuickPi
 	}
 
 	set items(items: Array<T | IQuickPickSeparator>) {
+		const shouldKeepFocus = this.shouldKeepFocus(items);
+		let scrollTopBefore: number | undefined;
+		if (shouldKeepFocus) {
+			const newActiveItems: T[] = [];
+			// Find the new object references of the same id.
+			for (const activeItem of this._activeItems) {
+				const item = items.find(i => (i as T).id === activeItem.id);
+				if (item) {
+					newActiveItems.push(item as T);
+				}
+			}
+
+			this._activeItems = newActiveItems;
+			this.activeItemsUpdated = true;
+			scrollTopBefore = (this.ui.list as any).list.scrollTop;
```

Comment:
what would be the best way to get this? Add a `scrollTop` prop to `this.ui.list`?